java/src/Ice/src/main/java/com/zeroc/Ice/ObjectAdapter.java

```diff
@@ -1449,57 +1431,81 @@ private List<EndpointI> parseEndpoints(String endpts, boolean oaEndpoints) {
     }
 
     private EndpointI[] computePublishedEndpoints() {
-        List<EndpointI> endpoints;
+        EndpointI[] endpointsArray;
         if (_routerInfo != null) {
-            //
             // Get the router's server proxy endpoints and use them as the published endpoints.
-            //
-            endpoints = new ArrayList<>();
-            for (EndpointI endpt : _routerInfo.getServerEndpoints()) {
-                if (!endpoints.contains(endpt)) {
-                    endpoints.add(endpt);
-                }
-            }
+            endpointsArray = _routerInfo.getServerEndpoints();
         } else {
-            //
-            // Parse published endpoints. If set, these are used in proxies
-            // instead of the connection factory Endpoints.
-            //
-            String endpts =
-                    _instance
-                            .initializationData()
-                            .properties
-                            .getProperty(_name + ".PublishedEndpoints");
-            endpoints = parseEndpoints(endpts, false);
-            if (endpoints.isEmpty()) {
-                //
+            // Parse published endpoints. If set, these are used instead of the connection factory
+            // endpoints.
+            var endpointsList =
+                    parseEndpoints(
+                            _instance
+                                    .initializationData()
+                                    .properties
+                                    .getProperty(_name + ".PublishedEndpoints"),
+                            false);
+
+            if (endpointsList.isEmpty()) {
```

Comment:
I kept the same logic/flow as C++ and C#. One difficulty with Java is there is no non-terminal way to figure out if a stream is empty. This is why there are so many "collect" in this code.

Reply:
Seems you can do `findAny().isPresent()`

Reply:
That's what I did initially. The problem is findAny() is terminal, which means you can't use the stream afterwards.